lib/puppet/provider/user/useradd.rb

```diff
@@ -152,6 +152,37 @@ def groups=(value)
     set(:groups, value)
   end
 
+  def password=(value)
+    user = @resource[:name]
+    tempfile = Tempfile.new('puppet', :encoding => Encoding::UTF_8)
+    begin
+      # Puppet execute does not support strings as input, only files.
+      # The password is expected to be in an encrypted format given -e is specified:
+      tempfile << "#{user}:#{value}\n"
+      tempfile.flush
+
+      # Options '-e' use encrypted password
+      # Must receive "user:enc_password" as input
+      # command, arguments = {:failonfail => true, :combine => true}
+      cmd = [command(:chpasswd), '-e']
+      execute_options = {
+        :failonfail => false,
+        :combine => true,
+        :stdinfile => tempfile.path,
+        :sensitive => has_sensitive_data?
+      }
+      output = execute(cmd, execute_options)
+
+      # chpasswd can return 1, even on success (at least on AIX 6.1); empty output
+      # indicates success
+      raise Puppet::ExecutionFailure, "chpasswd said #{output}" if output != ''
```

Comment:
We raise `Puppet::ExecutionFailure` rescue it and then repackage it as an  `Puppet::Error`. But since ExecutionFailure` already is an instance of `Puppet::Error`, maybe just raise it outside of the begin/rescue?